packages/playwright-core/src/server/registry/index.ts

```diff
@@ -1248,9 +1251,29 @@ export class Registry {
     }
   }
 
-  private async _validateInstallationCache(linksDir: string) {
-    // 1. Collect used downloads and package descriptors.
-    const usedBrowserPaths: Set<string> = new Set();
+  async list() {
+    const linksDir = path.join(registryDirectory, '.links');
+    const browsers: Array<{ browserName: string, browserVersion: number, hostDir: string, browserPath: string }> = await this._traverseBrowserInstallations(linksDir);
+
+    // Group browsers by browserName
+    const groupedBrowsers: Record<string, Array<{ browserVersion: number, hostDir: string, browserPath: string }>> = {};
+
+    browsers.forEach(browser => {
+      if (!groupedBrowsers[browser.browserName])
+        groupedBrowsers[browser.browserName] = [];
+
+      groupedBrowsers[browser.browserName].push({
+        browserVersion: browser.browserVersion,
+        hostDir: browser.hostDir,
+        browserPath: browser.browserPath
+      });
+    });
+
+    return groupedBrowsers;
+  }
+
+  private async _traverseBrowserInstallations(linksDir: string) {
+    const browserList: Array<{ browserName: string, browserVersion: number, hostDir: string, browserPath: string }> = [];
```

Comment:
`{ browserName: string, browserVersion: number, hostDir: string, browserPath: string }` is repeated several times. Let's introduce a named type for it, something like `InstalledBrowserInfo`.